### ISSUE TITLE
Removing ordered columns from grouping result panel generates wrong SQL query

### ISSUE THREAD

**Issue by Vichman67** [dbeaver/dbeaver#19406]
### Description

Hello,

This bug affects the behaviour of DBeaver when using Oracle driver. I've checked another drivers (MySQL, SQL Server) without beeing able to reproduce it. Furthermore, in the Oracle server where I got this error, it affected only to certain tables, and not to all of them (without any differences in the affected and non-affected tables in terms of DDL neither number of records).

When using results panel for grouping data in tables data tab you can sort the grouping results by any of the columns included.

In the following video I perform the following operations:
1. Open the results panel.
2. Select the gouping tab.
3. Group data by AEDTM column.
4. Add additional grouping column UNAME.
5. Sort the group results by one of the columns (AEDTM).

Then, if you remove one of the grouping columns (e.g. AEDTM), provided the removed column has been previously used in ordering, DBeaver generates a wrong GROUP BY query, cause it keeps the ORDER BY based on the removed column. 

This is the wrong query generated:

```
SELECT UNAME, COUNT(*) FROM ODS_RH.TP_ST_HRP1000 x
GROUP BY UNAME
ORDER BY AEDTM
```

It keeps 'ORDER BY AEDTM' but AEDTM has been removed and is not in the SELECT statement.

You can see the steps in the following video.

https://user-images.githubusercontent.com/12735950/225865705-115a5134-8fad-4d55-b843-c6dd8a615932.mp4

Thanks!

### DBeaver Version

Enterprise 23.0.0

### Operating System

Windows 10 Education 64-bit

### Database and driver

Database: Oracle Database 12c Enterprise Edition Release 12.2.0.1.0 - 64bit Production
Driver: Oracle

**Comment by Rekkonnect** [comment]
Could reproduce this, but only the first time I opened any grouping window. When I introduced the new column by dragging it into the grouping panel, it was placed first. All the other times that I tried reproducing this, it appeared last right before the COUNT(*) column. No matter which column I tried sorting and then removing, the bug could not be reproduced, even on the same table that triggered the bug altogether.
